gno.land/pkg/gnoland/app.go

```diff
@@ -196,9 +225,109 @@ func InitChainer(baseApp *sdk.BaseApp, acctKpr auth.AccountKeeperI, bankKpr bank
 	}
 }
 
-// XXX not used yet.
-func EndBlocker(vmk vm.VMKeeperI) func(ctx sdk.Context, req abci.RequestEndBlock) abci.ResponseEndBlock {
-	return func(ctx sdk.Context, req abci.RequestEndBlock) abci.ResponseEndBlock {
-		return abci.ResponseEndBlock{}
+// endBlockerApp is the app abstraction required by any EndBlocker
+type endBlockerApp interface {
+	// LastBlockHeight returns the latest app height
+	LastBlockHeight() int64
+
+	// Logger returns the logger reference
+	Logger() *slog.Logger
+}
+
+// EndBlocker defines the logic executed after every block.
+// Currently, it parses events that happened during execution to calculate
+// validator set changes
+func EndBlocker(
+	collector *collector[validatorUpdate],
+	vmKeeper vm.VMKeeperI,
```

Comment:
```suggestion
	vmk vm.VMKeeperI,
```

To maintain consistency for this object across the codebase, it's best to treat it as if it were a singleton.

Reply:
Swapped, in other places too:

[2592b43](https://github.com/gnolang/gno/pull/2229/commits/2592b43d21ad55b1039253b2f8adade9778b12a4)